dt-workflows/workflows-execution-handler.php

```diff
@@ -583,14 +583,87 @@ private static function action_append( $field_type, $field_id, $value ): array {
                 }
                 break;
             case 'comments':
+
+                // get post settings of the current post
+                $settings = DT_Posts::get_post_settings( $post['post_type'] );
+                // filter down to only reference fields - or can you just get connection_types?
+                $fields = array_filter( $settings['fields'], function( $obj ) {
+                    if ( $obj['type'] == 'connection' || $obj['type'] == 'user_select' ) {
+                        return true;
+                    }
+                    return false;
+                } );
+
+                // loop through each field, checking if the comment attempts to reference one of the fields
+                foreach ( $fields as $field => $data ) {
+                    if ( str_contains( $value, '{' . $field . '}' ) ) {
+                        $replacement = '';
+                        $current = 0;
+                        $references = $post[$field]; // get array of references from the field
+
+                        // special case: if the field is 'assigned_to', it is not an array of references,
+                        // but an array of the pieces of the 'assigned_to' field
+                        if ( $field == 'assigned_to' ) {
```

Comment:
Instead of using the field name, can we use the field type of `user_select`? People can add multiple user select fields if they so choose.

Reply:
Sounds good to me. The issue was assigned_to only referenced a single record, so it wasn't an array of records. If people add multiple user_select fields, would they also only be able to reference a single user? If so then this would work, otherwise we'll still need a special case for 'assigned_to'

Reply:
Yeah, all user_select records only allow selecting a single user, so they will all act the same.